core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java

```diff
@@ -142,18 +145,16 @@ Status tryHandleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
       }
 
       LoadBalancer delegate = getDelegate();
-      if (resolvedAddresses.getAddresses().isEmpty()
-          && !delegate.canHandleEmptyAddressListFromNameResolution()) {
+      if (delegate.handleResolvedAddresses(
+          ResolvedAddresses.newBuilder()
+              .setAddresses(resolvedAddresses.getAddresses())
+              .setAttributes(attributes)
+              .setLoadBalancingPolicyConfig(lbConfig)
+              .build())) {
+        return Status.OK;
+      } else {
         return Status.UNAVAILABLE.withDescription(
             "NameResolver returned no usable address. addrs=" + servers + ", attrs=" + attributes);
```

Comment:
Looks like this string needs to be updated. But also, this architecture will need to change because this status will [override the status](https://github.com/grpc/grpc-java/blob/8e7793652b21e316740cb0ff5f940a0987d3e679/core/src/main/java/io/grpc/internal/ManagedChannelImpl.java#L1871) that the LB policy is suppose to be providing via the Picker. If this PR is "stage one" and there will be some other changes coming, that is fine.

Reply:
I updated the string, but is this what you had in mind? I did not make the additional changes you mentioned because this PR is already very large and I want to better understand what to do.